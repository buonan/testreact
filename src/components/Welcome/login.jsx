import React, { useState } from 'react';

const Login = (props) => {
  const [isLoggedIn, setLoggedIn] = useState(false)
  const [isMounted, setMounted] = useState(false)

  // Note: React.useEffect vs useEffect (will not use imported mock)
  React.useEffect(() => {
    setMounted(true)
  })

  const handleClick = () => {
    setLoggedIn(true)
  }

  return (
    <div>
      <h1>Login</h1>
      <button onClick={handleClick}></button>
    </div>
  )
}

export default Login