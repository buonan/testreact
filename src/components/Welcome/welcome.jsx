import React from 'react'

class Welcome extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isClicked: false,
      isMounted: false
    }
    this.handleClick = this.handleClick.bind(this)
  }
  componentDidMount() {
    this.setState(state => ({
      isMounted: true
    }))
  }
  handleClick() {
    this.setState(state => ({
      isClicked: true
    }))
    console.log('Clicked!')
  }
  render() {
    return (
      <div>
        <h1>Hello, {this.props.name}</h1>
        <button onClick={this.handleClick}></button>
      </div>
    );
  }
}

export default Welcome