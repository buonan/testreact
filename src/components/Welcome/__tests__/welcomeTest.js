import React from 'react'
import { mount, shallow, render } from 'enzyme'
import Welcome from '../welcome'

describe('Welcome Component tests', ()=> {
  it('state test', ()=>{
    const component = mount(<Welcome />)
    expect(component.find('Welcome').exists()).toBe(true)
    expect(component.state().isMounted).toBe(true)
    expect(component.state().isClicked).toBe(false)

    const b = component.find('button')
    b.simulate('click')
    expect(component.state().isClicked).toBe(true)
  })

  it('props test', ()=>{
    const component = mount(<Welcome name='Brian' />)
    expect(component.find('Welcome').exists()).toBe(true)
    expect(component.props().name).toBe('Brian')
  })

  it('should match snapshot and styles for default props', ()=>{
    const component = mount(<Welcome/>)
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot and styles for default props', ()=>{
    const component = render(<Welcome/>)
    console.log(component)
  })
})