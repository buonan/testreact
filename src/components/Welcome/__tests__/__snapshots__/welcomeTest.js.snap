// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component tests should match snapshot and styles for default props 1`] = `
<Welcome>
  <div>
    <h1>
      Hello, 
    </h1>
    <button
      onClick={[Function]}
    />
  </div>
</Welcome>
`;
